File: Lab 8/Java/Problem4.java
import java.util.*;
import java.io.*;

public class Problem4 {
    static class Task implements Comparable<Task> {
        int s, e;

        Task(int s, int e) {
            this.s = s;
            this.e = e;
        }

        public int compareTo(Task other) {
            return Integer.compare(this.e, other.e);
        }
    }

    public static void main(String[] args) throws IOException {
        Scanner sc = new Scanner(System.in);
        if (!sc.hasNextInt())
            return;
        int N = sc.nextInt();
        Task[] tasks = new Task[N];
        for (int i = 0; i < N; i++) {
            tasks[i] = new Task(sc.nextInt(), sc.nextInt());
        }
        Arrays.sort(tasks);

        List<Task> res = new ArrayList<>();
        long last = Long.MIN_VALUE;

        for (Task t : tasks) {
            if (t.s > last) {
                res.add(t);
                last = t.e;
            }
        }

        System.out.println(res.size());
        for (Task t : res) {
            System.out.println(t.s + " " + t.e);
        }
    }
}

File: Lab 8/Problem4.py
import sys
input = sys.stdin.readline

N = int(input())
tasks = []

for _ in range(N):
    s,e = map(int, input().split())
    tasks.append((e,s))

tasks.sort()

res = []
last = -10**18

for e,s in tasks:
    if s > last:
        res.append((s,e))
        last = e

print(len(res))
for s,e in res:
    print(s,e)
